FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -278,35 +277,28 @@ final class MessageService {
 
 // MARK: - Message verification
 extension MessageService {
-    private func fetchVerificationPubKeys(for sender: String?) -> [String] {
-        if let sender = sender {
-            return contactsService.retrievePubKeys(for: sender)
-        } else {
-            return []
-        }
+    private func fetchVerificationPubKeys(for sender: String?) async throws -> [String] {
+        guard let sender = sender else { return [] }
+
+        let pubKeys = contactsService.retrievePubKeys(for: sender)
+        if pubKeys.isNotEmpty { return pubKeys }
+
+        guard let contact = try? await contactsService.searchContact(with: sender)
+        else { return [] }
+
+        return contact.activePubKeys.map(\.armored)
```

Comment:
I'm not too sure about the .activePubkeys here. I'd give it all of them. I guess, an expired public key is still a good public key to verify a message that was made before the key expired. But that's for TypeScript to decide, not here.

Reply:
Also, by also giving it revoked pubkeys, we give TypeScript a chance to say the signature is bad. But if we withold the revoked pubkeys, TypeScript will say that the right pubkey is missing - which is different.

Reply:
Got it, I'll change it to `contact.pubKeys.map(\.armored)`